modules/BusinessActions/BusinessActions.php

```diff
@@ -299,9 +299,25 @@ public static function getAllByType($tabid, $type = false, $parameters = false,
 			}
 
 			if ($parameters) {
-				$link->linkurl = $strtemplate->merge($link->linkurl);
-				$link->linkicon= $strtemplate->merge($link->linkicon);
-			}
+                $link->linkurl = $strtemplate->merge($link->linkurl);
+                $link->linkicon= $strtemplate->merge($link->linkicon);    
+
+				if (!empty($parameters['RECORD'])) {
+					$focus_bs = CRMEntity::getInstance($parameters['MODULE']);
+					$focus_bs->retrieve_entity_info($parameters['RECORD'], $parameters['MODULE']);
+					$data_retrieve = $focus_bs->column_fields;
+					$str = $link->linkurl;
+					$subtring_start = strpos($str, '$'); 
+					$subtring_start += strlen('$');   
+					$size = strpos($str, '$', $subtring_start) - $subtring_start;   
+					$substring = substr($str, $subtring_start, $size);
+					$match = str_replace('$'.$substring.'$', $data_retrieve[$substring], $link->linkurl);
+					include_once 'include/Webservices/Create.php';
+					$entityCache = new VTEntityCache($current_user);
+					$ct = new VTSimpleTemplate($match, false);
+					$link->linkurl = $ct->render($entityCache, vtws_getEntityId($parameters['MODULE']).'x'.$parameters['RECORD']);
+				}
+            }
 			if ($multitype) {
```

Comment:
Correctly format the code
Note that VTSimpleTemplate uses the workflow syntax, not the limited format of the StringTemplate class, so you will have to test with the workflow field syntax.
Please create a new PR